src/common/utxobased/keymanager/bip69.ts

```diff
@@ -0,0 +1,54 @@
+// Previous transaction hashes (in reversed byte-order) are to be sorted in ascending order, lexicographically.
+// In the event of two matching transaction hashes, the respective previous output indices will be compared by their integer value, in ascending order.
+
+import { Input, Output } from './utxopicker'
+
+// If the previous output indices match, the inputs are considered equal.
+const inputComparator = (
+  aHash: string | Buffer,
+  aVout: number,
+  bHash: string | Buffer,
+  bVout: number
+): number => {
+  if (typeof aVout !== 'number' || typeof bVout !== 'number')
+    throw new TypeError('Expected vouts of type Number')
+  if (typeof aHash === 'string' && typeof bHash === 'string') {
+    const compare = aHash.localeCompare(bHash)
+    return compare !== 0 ? compare : aVout - bVout
+  }
+
+  if (!Buffer.isBuffer(aHash) || !Buffer.isBuffer(bHash))
+    throw new TypeError('Expected hashes of type Buffer')
+
+  const aHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
+  const bHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
+  const compare = aHashR.compare(bHashR)
+  return compare !== 0 ? compare : aVout - bVout
+}
```

Comment:
nit:

```ts
const inputComparator = (
  aHash: string | Buffer,
  aVout: number,
  bHash: string | Buffer,
  bVout: number
): number => {
  if (typeof aVout !== 'number' || typeof bVout !== 'number')
    throw new TypeError('Expected vouts of type Number')
  
  const compare = (() => {
    if (typeof aHash === 'string' && typeof bHash === 'string') {
      return aHash.localeCompare(bHash)
    }
    if (Buffer.isBuffer(aHash) && Buffer.isBuffer(bHash)){  
      const aHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
      const bHashR = ([].reverse.call(Buffer.from(aHash)) as unknown) as Buffer
      return aHashR.compare(bHashR)
    }
    throw new TypeError('Expected hashes of type Buffer or string')
  })()

  return compare !== 0 ? compare : aVout - bVout
}
```

What you think?

Reply:
I rewrote it without all the dynamic type checking and it's much simpler now.